split_vocab.py

```
@@ -0,0 +1,41 @@
+import argparse
+
+
+def split_file(input_file, output_prefix, programs_per_file):
+    with open(input_file, 'r') as f:
+        current_file = 1
+        current_program_count = 0
+        current_program = []
+        output = open(f"{output_prefix}_{current_file}.txt", 'w')
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)